crates/marketplace/src/service.rs

```diff
@@ -508,13 +547,34 @@ where
             txns.iter().map(|txn| txn.commit()).collect::<Vec<_>>()
         );
         let txns = self.hooks.process_transactions(txns).await;
+        for txn in txns.clone() {
+            let _ = self
+                .global_state
+                .write_arc()
+                .await
+                .set_tx_status(txn.commit(), TransactionStatus::Pending);
+        }
         let response = self
             .global_state
             .read_arc()
             .await
-            .submit_client_txns(txns)
+            .submit_client_txns(txns.clone())
             .await;
 
+        let pairs: Vec<(<Types as NodeType>::Transaction, Result<_, _>)> = (0..txns.len())
+            .map(|i| (txns[i].clone(), response[i].clone()))
+            .collect();
+        for (txn, res) in pairs {
+            if let Err(some) = res {
+                let _ = self.global_state.write_arc().await.set_tx_status(
+                    txn.commit(),
+                    TransactionStatus::Rejected {
```

Comment:
We should also update txn status when handling public mempool transactions in `run_builder_service`

Reply:
Right! There are two places we mark txn status as Rejected. One public pool and one private pool. Or is there the other place we `run_builder_service` beside `run_non_permissioned_standalone_builder_service`?